File: programmers/level2/프로세스.py
def solution(priorities, location):
    answer = 0
    idx = 0
    l = len(priorities)
    while True:
        if priorities[idx] == max(priorities[idx:]):
            if idx%l == location:
                return answer+1
            idx += 1
            answer += 1
            priorities.append(0)
        else:
            priorities.append(priorities[idx])
            idx += 1

    return answer

priorities = [1, 1, 9, 1, 1, 1]
location = 0

print(solution(priorities, location))

File: 프로그래머스/2/42587. 프로세스/프로세스.java
import java.util.*;
class Position {
    int pr;
    int idx;
    Position(int a, int b) {
        pr=a; idx=b;
    }
}

class Solution {
    public int solution(int[] priorities, int location) {
        int answer = 1;
        Queue<Position> q = new LinkedList<>();
        for(int i=0; i<priorities.length; i++) {
            q.offer(new Position(priorities[i], i));
        }
        
        int max = Integer.MIN_VALUE;
        boolean flag = false;
        while(true) {
            //최대값 구해두기
            if(!flag) {
                for(Position n:q) {
                    max = Math.max(max, n.pr);
                }
                flag = true;
            }
            Position p = q.poll();
            if(p.idx==location && max==p.pr) break;
            if(max==p.pr) {
                answer++;
                flag = false;
                max=Integer.MIN_VALUE;
            } else {
                q.offer(p);
            }
        }
        
        
        return answer;
    }
}